pkg/tsdb/loki/scopes_test.go

```diff
@@ -0,0 +1,72 @@
+package loki
+
+import (
+	"testing"
+
+	"github.com/grafana/grafana/pkg/promlib/models"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInjectScopesIntoLokiQuery(t *testing.T) {
+	tests := []struct {
+		name         string
+		query        string
+		scopeFilters []models.ScopeFilter
+		expected     string
+		expectErr    bool
+	}{
+		{
+			name:      "No filters with no existing filter",
+			query:     `{} |= "an unexpected error"`,
+			expected:  `{} |= "an unexpected error"`,
+			expectErr: false,
+		},
+		{
+			name:      "No filters with existing filter",
+			query:     `{namespace="default"} |= "an unexpected error"`,
+			expected:  `{namespace="default"} |= "an unexpected error"`,
+			expectErr: false,
+		},
+		{
+			name:  "scopes with existing filter",
+			query: `{namespace="default"} |= "an unexpected error"`,
+			scopeFilters: []models.ScopeFilter{
+				{Key: "cluster", Value: "us-central-1", Operator: models.FilterOperatorEquals},
+			},
+			expected:  `{namespace="default", cluster="us-central-1"} |= "an unexpected error"`,
+			expectErr: false,
+		},
+		{
+			name:  "scopes without existing label matchers",
+			query: `{} |= "an unexpected error"`,
+			scopeFilters: []models.ScopeFilter{
+				{Key: "cluster", Value: "us-central-1", Operator: models.FilterOperatorEquals},
+			},
+			expected:  `{cluster="us-central-1"} |= "an unexpected error"`,
+			expectErr: false,
+		},
+		{
+			name:  "scopes with multiple filters",
+			query: `{} |= "an unexpected error"`,
```

Comment:
Can we also test something with multiple matchers? And for sanity, a metrics query would be good.
Something like `count_over_time({} [1m])` and `count_over_time({} |= "error" [1m])/count_over_time({} [1m])`.

Reply:
I think the current implementation in https://github.com/grafana/grafana/pull/95591/files#diff-216743b3d57c4b78d2ab141e972f3c6b01f726647dd8113073dc81c1d695892dR21-R28 would already handle the cases I was talking about. Was just suggesting adding those as additional test cases, maybe.

Reply:
👍